src/set.cr

```diff
@@ -37,6 +37,11 @@ struct Set(T)
     @hash = Hash(T, Nil).new(initial_capacity: initial_capacity)
   end
 
+  # Optimized version of `new` used when `other` is also an `Indexable`
```

Comment:
Small correction: `other` in this case is a parameter, so use asterisks to wrap instead of backticks `*other*`.

Mode details on the documentation style here: https://crystal-lang.org/docs/conventions/documenting_code.html